src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -161,14 +161,20 @@ export default App.extend({
 
     channel.request('subscribe', this.collection.models, { filters: { [filterKey]: this.filters } });
 
-    this.listenTo(channel, 'message', (message, model) => {
+    this.listenTo(channel, 'message', ({ category, payload }) => {
+      if (category !== 'ResourceCreated') return;
+
+      const model = Radio.request('entities', 'get:store', payload.resource);
+
       if (this.collection.get(model) || model.isFetching()) return;
 
       model.fetch().then(() => {
         // notifications do not fully filter our action flow_states
         if (filterKey === 'actions' && !flowStates.includes(model.getFlow().getState().id)) return;
 
         this.collection.add(model);
```

Comment:
Do we need to do anything with the `this.filteredCollection` and `this.editableCollection` here?

I haven't looked into the details yet to answer that question.